src/scheduler/node_partition.c

```diff
@@ -423,6 +423,9 @@ create_node_partitions(status *policy, node_info **nodes, char **resnames, unsig
 				res = &unset_res;
 			}
 			if (res != NULL) {
+				/* Incase of indirect resource, point it to the right place */
```

Comment:
im sure there's good reason to do it this way, i'm just curious, why don't we just make 'res' be 'indirect_res' when it is an indirect resource?

Reply:
IMO it is because there will be multiple node objects pointing to the same object and when you free them you will have dangling pointers all over the place. Unless we decide to add a reference count to sched_resource structure.